runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/ast/CachingArtifactAstDependencyGraphProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ */
+package org.mule.runtime.config.internal.validation.ast;
+
+import static com.github.benmanes.caffeine.cache.Caffeine.newBuilder;
+
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.graph.api.ArtifactAstDependencyGraph;
+import org.mule.runtime.ast.graph.api.ArtifactAstDependencyGraphFactory;
+import org.mule.runtime.ast.graph.api.ArtifactAstDependencyGraphProvider;
+
+import com.github.benmanes.caffeine.cache.Cache;
+
+
+/**
+ * An {@link ArtifactAstDependencyGraphProvider} that caches the {@link ArtifactAstDependencyGraphProvider}.
+ *
+ * TODO W-13931931: verify if it makes sense to have a bean in the registry to be used apart from the use case for AST validations
```

Comment:
registry won't be good for this case since it will cause the depGraph and the AST to be held in memory for as long as the app is tunning

Reply:
It also applies to the other cases where we talk where it is already been held.
I've changed the TODO in the javadoc.